src/ComputeSharp.D2D1/Shaders/Interop/Effects/TransformMappers/D2D1DrawInfoUpdateContextImpl.ID2D1DrawInfoUpdateContex.cs

```diff
@@ -0,0 +1,166 @@
+
+        /// <inheritdoc cref="SetConstantBuffer"/>
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        public delegate int SetConstantBufferDelegate(D2D1DrawInfoUpdateContextImpl* @this, byte* buffer, uint bufferCount);
+
+        /// <summary>
+        /// A cached <see cref="QueryInterfaceDelegate"/> instance wrapping <see cref="QueryInterface"/>.
+        /// </summary>
+        public static readonly QueryInterfaceDelegate QueryInterfaceWrapper = QueryInterface;
+
+        /// <summary>
+        /// A cached <see cref="AddRefDelegate"/> instance wrapping <see cref="AddRef"/>.
+        /// </summary>
+        public static readonly AddRefDelegate AddRefWrapper = AddRef;
+
+        /// <summary>
+        /// A cached <see cref="ReleaseDelegate"/> instance wrapping <see cref="Release"/>.
+        /// </summary>
+        public static readonly ReleaseDelegate ReleaseWrapper = Release;
+
+        /// <summary>
+        /// A cached <see cref="GetConstantBufferSizeDelegate"/> instance wrapping <see cref="GetConstantBufferSize"/>.
+        /// </summary>
+        public static readonly GetConstantBufferSizeDelegate GetConstantBufferSizeWrapper = GetConstantBufferSize;
+
+        /// <summary>
+        /// A cached <see cref="GetConstantBufferDelegate"/> instance wrapping <see cref="GetConstantBuffer"/>.
+        /// </summary>
+        public static readonly GetConstantBufferDelegate GetConstantBufferWrapper = GetConstantBuffer;
+
+        /// <summary>
+        /// A cached <see cref="SetConstantBufferDelegate"/> instance wrapping <see cref="SetConstantBuffer"/>.
+        /// </summary>
+        public static readonly SetConstantBufferDelegate SetConstantBufferWrapper = SetConstantBuffer;
+#endif
+
+        /// <inheritdoc cref="D2D1DrawInfoUpdateContextImpl.QueryInterface"/>
+        [UnmanagedCallersOnly]
+        public static int QueryInterface(D2D1DrawInfoUpdateContextImpl* @this, Guid* riid, void** ppvObject)
+        {
+            return @this->QueryInterface(riid, ppvObject);
+        }
+
+        /// <inheritdoc cref="D2D1DrawInfoUpdateContextImpl.AddRef"/>
+        [UnmanagedCallersOnly]
+        public static uint AddRef(D2D1DrawInfoUpdateContextImpl* @this)
+        {
+            return @this->AddRef();
+        }
+
+        /// <inheritdoc cref="D2D1DrawInfoUpdateContextImpl.Release"/>
+        [UnmanagedCallersOnly]
+        public static uint Release(D2D1DrawInfoUpdateContextImpl* @this)
+        {
+            return @this->Release();
+        }
+
+        /// <inheritdoc cref="ID2D1DrawInfoUpdateContext.GetConstantBufferSize"/>
+        [UnmanagedCallersOnly]
+        public static int GetConstantBufferSize(D2D1DrawInfoUpdateContextImpl* @this, uint* size)
+        {
+            if (@this->d2D1DrawInfo is null)
+            {
+                return RO.RO_E_CLOSED;
+            }
+
+            if (size is null)
```

Comment:
parameter validation should generally be the first thing you check

Reply:
as in, first check the method parameters, _then_ check internal object state. If someone passes `null` for `size`, they should immediately get an `E_POINTER`

Reply:
Fixed in 27987efd028d1184f3e44e8636e8c116225aa370 👍